test/integration_test.py

```diff
@@ -386,13 +386,23 @@ def test_catalog_listing(self):
 
     def test_snapshot_listing(self):
         """
-        Test with a small page size to be sure paging works
+        Test with small page sizes to make sure paging works.
         """
-        page_size = 5
-        with mock.patch.object(TDRClient, 'page_size', page_size):
-            paged_snapshots = self._public_tdr_client.snapshot_names_by_id()
-        snapshots = self._public_tdr_client.snapshot_names_by_id()
-        self.assertEqual(snapshots, paged_snapshots)
+        # Without a filter, the test takes so long that there's a real risk of
+        # failure due to new snapshots being added mid-test.
+        #
+        snapshot_filters_by_deployment = {
+            'dev': 'hca_dev_5',
+            'anvildev': 'anvil_',
+            'prod': '_dcp37',
+            'anvilprod': 'anvil_page_'
+        }
+        filter = snapshot_filters_by_deployment[config.main_deployment_stage]
+        for page_size in 1, 2:
```

Comment:
There should be sub test for each iteration of this loop.

I also think we should assert the number of pages to be greater than 1, and that `len(snapshots)` is below a certain limit. I forgot what actual result sizes we observed during our recent PL slot but I think you could propose a limit and explain your thought process for that in a PR comment.

Reply:
I think you missed the 

> I also think we should assert the number of pages to be greater than 1

part